packages/components/src/SquigglePlayground.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { FC, useState } from "react";
+import ReactDOM from "react-dom";
+import { SquiggleChart } from "./SquiggleChart";
+import CodeEditor from "./CodeEditor";
+import { Form, Input, Card, Row, Col } from "antd";
+import "antd/dist/antd.css";
+
+interface FieldFloatProps {
+  label: string;
+  className?: string;
+  value: number;
+  onChange: (value: number) => void;
+}
+
+function FieldFloat(Props: FieldFloatProps) {
+  let [contents, setContents] = useState(Props.value + "");
+  return (
+    <Form.Item label={Props.label}>
+      <Input
+        value={contents}
+        className={Props.className ? Props.className : ""}
+        onChange={(e) => setContents(e.target.value)}
+        onBlur={(_) => {
+          let result = parseFloat(contents);
+          if (result != NaN) {
+            Props.onChange(result);
+          }
+        }}
+      />
+    </Form.Item>
+  );
+}
+
+interface Props {
+  initialSquiggleString: string;
+}
+
+let SquigglePlayground: FC<Props> = (props) => {
+  let [squiggleString, setSquiggleString] = useState(
+    props.initialSquiggleString
+  );
+  let [sampleCount, setSampleCount] = useState(1000);
+  let [outputXYPoints, setOutputXYPoints] = useState(1000);
+  let [pointDistLength, setPointDistLength] = useState(1000);
+  let [diagramStart, setDiagramStart] = useState(0);
+  let [diagramStop, setDiagramStop] = useState(10);
+  let [diagramCount, setDiagramCount] = useState(20);
+  var demoDist = (
+    <SquiggleChart
+      squiggleString={squiggleString}
+      sampleCount={sampleCount}
+      outputXYPoints={outputXYPoints}
+      diagramStart={diagramStart}
+      diagramStop={diagramStop}
+      diagramCount={diagramCount}
+      pointDistLength={pointDistLength}
+    />
+  );
+  return (
+    <Row>
+      <Col span={12}>
+        <Card title="Distribution Form">
+          <Form>
+            <Row gutter={16}>
+              <Col span={24}>
+                <CodeEditor
+                  value={squiggleString}
+                  onChange={setSquiggleString}
+                  oneLine={false}
+                />{" "}
+              </Col>
+            </Row>
+            <Row gutter={16}>
+              <Col span={12}>
+                <FieldFloat
+                  value={sampleCount}
+                  label="Sample Count"
+                  onChange={setSampleCount}
+                />{" "}
```

Comment:
Whats with the empty string?

Reply:
Not sure? I think it's from your react code

Reply:
Ah; in that case it probably had some function. I forgot what it was though.